source/Irrlicht/COpenGLDriver.cpp

```diff
@@ -1046,7 +1046,8 @@ bool COpenGLDriver::genericDriverInit()
 	///extGlProvokingVertex(GL_FIRST_VERTEX_CONVENTION_EXT);
 
 	// create material renderers
-	createMaterialRenderers();
+	//createMaterialRenderers();
+    //WARNING EMT_SOLID and other "default shaders" won't work now
```

Comment:
nuke E_MATERIAL_TYPE, but add the old 4 default shader combos as IGPUMaterials to IAssetManager's CPUtoGPU object cache and add associated ICPU* objects for them

Reply:
now after we decided to do pipelines etc, they should be 4x IGPUGraphicPipelines. right?

Reply:
@Crisspl TODO

Reply:
status?

Reply:
none of this exist yet, will be done as one of last things

Reply:
all right, but lets keep them very separate from all of the engine, maybe lets move `E_MATERIAL_TYPE` into `IVideoDriver` and have a function `getBuiltInSpecializedShader(E_SHADER_STAGE,E_MATERIAL_TYPE)`